src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/ValidationErrorMetadata.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
+{
+    public class ValidationErrorMetadata
+    {
+        /// <summary>
+        /// Error message the model binding system adds when a property with an associated
+        /// <see cref="BindRequiredAttribute"/> is not bound.
+        /// </summary>
+        /// <value>Default <see cref="string"/> is "A value for the '{0}' property was not provided.".</value>
+        public Func<string> MissingBindRequiredValueResource { get; set; }
```

Comment:
We need to make sure that these can't be set to null. It would be really hard to debug (for a user) if they did something silly here and got random 500s because of a nullref.